src/compiler/emitter.ts

```diff
@@ -491,20 +497,120 @@ module ts {
                     }
                 }
 
+                function convertToBase64(input: string): string {
+                    // Encode a string into Base64 to include it in a URI for instance.
+                    // Effectively what we are doing here is putting every 3 charachters 
+                    // into 4 base64 digits.
+                    // UTF-8 makes this a bit tricky as each charater may need to be expanded
+                    // to 1, 2 or 3 charaters to ensure the conversion is correct.
+                    // To do that we consider one character from the input at a time, for non-ASCII 
+                    // character expand them into 2 or 3 as appropriate. Then take 3 expanded chars
+                    // at a time and convert them into 4 base-64 digits and add them to the ouput.
+
+                    // Output
+                    var result = "";
+
+                    // A working set of charachter points to write in base64 encoding. We only need
+                    // 3 character points at a time, so the most number of points we can have is 5.

```

Comment:
not sure what kind of example i can include.. it is all about the character value.